MonoGame.Framework/Graphics/Texture2D.cs

```diff
@@ -422,5 +423,166 @@ private static void ConvertToABGR(int pixelHeight, int pixelWidth, int[] pixels)
                                             "elementCount * sizeof(T) is {0}, but data size is {1}.",
                                             elementCount * tSize, dataByteSize), "elementCount");
         }
-	}
+
+        internal Color[] GetColorData()
+        {
+            int colorDataLength = Width * Height;
+            var colorData = new Color[colorDataLength];
+
+            switch (Format)
+            {
+                case SurfaceFormat.Single:
+                    var floatData = new float[colorDataLength];
+                    GetData(floatData);
+
+                    for (int i = 0; i < colorDataLength; i++)
+                    {
+                        float brightness = floatData[i];
+                        // Export as a greyscale image.
+                        colorData[i] = new Color(brightness, brightness, brightness);
+                    }
+                    break;
+
+                case SurfaceFormat.Color:
+                    GetData(colorData);
+                    break;
+
+                case SurfaceFormat.Alpha8:
+                    var alpha8Data = new Alpha8[colorDataLength];
+                    GetData(alpha8Data);
+
+                    for (int i = 0; i < colorDataLength; i++)
+                    {
+                        colorData[i] = new Color(((IPackedVector)alpha8Data[i]).ToVector4());

```

Comment:
Use new `Color(0, 0, 0, alpha8Data.PackedValue)` so you don't need ToVector4, the interface cast causes boxing. Similar fix for the other IPackedVectors.

It would be better if all the IPackedVectors had a `ToVector4` function that isn't an explicit interface implementation, so we can call the method directly on the struct to avoid boxing and the conversions here can just do `xData[i].ToVector4()`.

Reply:
The explicit interface implementation and normal ToVector4 method can exist side by side btw.

Reply:
Would be nice if these types are all readonly as well, so no shadow copying happens, but we can't do that because we're on an old C# version :/

Reply:
We can make ToVector4 public method, than it'll cover the interface method too.
That is what my new commit.